packages/components/src/utils/strings.ts

```diff
@@ -71,6 +72,36 @@ export const normalizeTextString = ( value: string ): string => {
 		.replace( ALL_UNICODE_DASH_CHARACTERS, '-' );
 };
 
+/**
+ * Converts any string to kebab case.
+ * Backwards compatible with Lodash's `_.kebabCase()`.
+ * Backwards compatible with `_wp_to_kebab_case()`.
+ *
+ * @see https://lodash.com/docs/4.17.15#kebabCase
+ * @see https://developer.wordpress.org/reference/functions/_wp_to_kebab_case/
+ *
+ * @param {string} str String to convert.
+ * @return {string} Kebab-cased string
```

Comment:
No need to add types to the JSDoc comments since we're using TypeScript

Reply:
Current eslint rules seem to require at least `@param` only if we have a function comment. If I delete all function comments, the lint error will also disappear, but I feel that the comment for this function is necessary.

![jsdoc-error](https://github.com/WordPress/gutenberg/assets/54422211/09dbb3ae-06a5-4d2d-b17b-10216a8bdc28)

Reply:
You should be able to avoid the linter errors by removing only the type info, while leaving everything else

```diff
diff --git a/packages/components/src/utils/strings.ts b/packages/components/src/utils/strings.ts
index a1d192942a..bb43a5e53a 100644
--- a/packages/components/src/utils/strings.ts
+++ b/packages/components/src/utils/strings.ts
@@ -80,8 +80,8 @@ export const normalizeTextString = ( value: string ): string => {
  * @see https://lodash.com/docs/4.17.15#kebabCase
  * @see https://developer.wordpress.org/reference/functions/_wp_to_kebab_case/
  *
- * @param {string} str String to convert.
- * @return {string} Kebab-cased string
+ * @param str String to convert.
+ * @return Kebab-cased string
  */
 export function kebabCase( str: unknown ) {
 	let input = str?.toString?.() ?? '';
@@ -102,9 +102,9 @@ export function kebabCase( str: unknown ) {
 /**
  * Escapes the RegExp special characters.
  *
- * @param {string} string Input string.
+ * @param string Input string.
  *
- * @return {string} Regex-escaped string.
+ * @return Regex-escaped string.
  */
 export function escapeRegExp( string: string ): string {
 	return string.replace( /[\\^$.*+?()[\]{}|]/g, '\\$&' );

```

Reply:
Ah, you mean remove only the type information. It seems I misunderstood 😅 Updated with [102e985](https://github.com/WordPress/gutenberg/pull/56758/commits/102e9851b1610a60c1b6b8b5944c511b68b8cd85).